module.nix

```diff
@@ -52,6 +56,22 @@ in
             for the associated Let's Encrypt certificate.
           '';
         };
+        vhost = mkOption {
+         type = types.nullOr (types.submodule
+          (import (modulesPath + "/services/web-servers/nginx/vhost-options.nix") {
+            inherit config lib;
+          }));
```

Comment:
Why duplicating the options here?

You can override all of that on your own by using `services.nginx.virtualHosts.${config.services.authentik.nginx.host}`.
If this isn't obvious enough, it should be documented somewhere.

The module system was designed to perform these kinds of overrides explicitly.

The current approach has the following downsides:
* dependency on the internal file structure of nixpkgs. Even though they're accessible from the outside, they're considered internal (and thus subject to change) by upstream. Example: https://github.com/NixOS/nixpkgs/pull/257100#discussion_r1376750403. This means as soon as something changes here, you'll need `if`/`else` hackery to support different nixpkgs revisions.
* dependency on the destructured arguments in `vhost-options.nix`. As soon as e.g. `pkgs` is used in there you'll run into the same issue.
* by providing these options, you'll need to manually ensure that all combinations of all options evaluate fine here (that's the reason for the long discussions above). When you only provide a small subset, it's up to the person deploying this module to fix their special cases (and it's a more difficult task here because you'll need to support different versions of nixpkgs while you can do all your customizations in your deployment for the nixpkgs rev you're currently using). Also, the module system is quite good at allowing you to override single options.
* when using `includeAllModules = true;` while building the `configuration.nix(8)` man-page (this gives you the option documentation for _all_ options available for your system config) you'll have the vhost options (and there are a few) duplicated (multiple times if others do that as well) which will blow up the documentation (and also make it harder to find the option you're interested in because you'll also have to check whether you're browsing the actual nginx options or the options of a module which duplicates a few - or all - of them).